ADC/Core/Src/init.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * init.c
+ *
+ *  Created on: Mar 10, 2020
+ *      Author: rodrigocelmer
+ */
+#include "stm32l053xx.h"
+
+void ADC_Init (void)
+{
+	/*
+	 * since we're going to initialize and then calibrate the ADC, we have to be sure that:
+	 * 1) ADEN = 0,
+	 * 2) DMAEN = 0,
+	 * 3) ADSTART = 0 and
+	 * 4) ADRDY = 0
+	 */
+	if(ADC1->CR & ADC_CR_ADSTART)			//checking if there's an ongoing conversion (3)
+	{
+		ADC1->CR |= ADC_CR_ADSTP;			//if so, we'll stop it
+		while(ADC1->CR & ADC_CR_ADSTP);
+	}
+	ADC1->CR |= ADC_CR_ADDIS;				//disables the ADC (1)
+	while(ADC1->CR & ADC_CR_ADEN);
+	ADC1->CFGR1 &= ~ADC_CFGR1_DMAEN;		//disables DMA (2)
+	if(ADC1->ISR & ADC_ISR_ADRDY)			//and we make sure ADRDY = 0 (4)
+	{
+		ADC1->ISR |= ADC_ISR_ADRDY;			//by setting it to 1
+	}
+
```

Comment:
As far as I remember, any of the operations that you are doing before line 30 are actually happening because you only enable ADC clock on RCC register on line 32.

Reply:
the operations are happening or are NOT happening? I couldn't understand. Should I enable RCC in line 11?

Reply:
not happening****